### PR TITLE
chore(storage): integ test for pageSize/nextToken in list()

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
Adds some integ tests for s3 list() API.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by HuiSF — APPROVED** [review]
Thanks for adding the tests.

File: packages/storage/amplify_storage_s3/example/integration_test/use_case_test.dart [hunk 1]
```diff
@@ -726,6 +726,99 @@ void main() {
...
+                options: StorageListOptions(
+                  accessLevel: accessLevel,
+                  pageSize: filesToList,
+                  nextToken: lastNextToken,
+                ),
+              ).result;
+              lastNextToken = listResult.nextToken;
+              timesCalled++;
+            }
+            expect(timesCalled, greaterThanOrEqualTo(filesToUpload));
```
Comment (dnys1) [inline packages/storage/amplify_storage_s3/example/integration_test/use_case_test.dart:812]: Can we not be more precise?

**Review by dnys1 — APPROVED** [review]

File: packages/storage/amplify_storage_s3/example/integration_test/use_case_test.dart [hunk 2]
```diff
@@ -726,6 +726,99 @@ void main() {
...
+                  metadata: {
+                    'filename': fileNameTemp,
+                  },
+                ),
+              ).result;
+            }
+
+            String? lastNextToken;
+            var timesCalled = 0;
+            while (lastNextToken != null || timesCalled == 0) {
```
[the remark this answers is not in the document]
Comment (ragingsquirrel3) [reply packages/storage/amplify_storage_s3/example/integration_test/use_case_test.dart]: good call, changed that

File: packages/storage/amplify_storage_s3/example/integration_test/use_case_test.dart (hunk 1 above)
Comment (ragingsquirrel3) [reply packages/storage/amplify_storage_s3/example/integration_test/use_case_test.dart]: changed to put in specific dir for this test and can be exact

**Review by HuiSF — APPROVED** [review]

**Review by dnys1 — APPROVED** [review]

**Pull request merged**